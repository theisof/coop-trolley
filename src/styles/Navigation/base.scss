$global-nav-height: 38px;
$site-nav-height: 60px;
$mobile-width: 80%;

@import './nav';
@import './global-nav';
@import './site-nav';
@import './nav-search';
@import './context-nav';

@media (max-width: $breakpoint-sm) {
  .coop-nav {
    .coop-context-nav,
    .coop-global-nav {
      transform: translate3d(-100%, 0, 0);
      transition: transform 0.3s;
    }
  }

  .coop-nav--mobile-visible {
    .coop-context-nav,
    .coop-global-nav {
      transform: translate3d(0, 0, 0);
    }
  }
}