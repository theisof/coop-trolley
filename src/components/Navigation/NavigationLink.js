import React from 'react'
import getImagePath from '../../utils/getImagePath'

const NavigationLink = ({ name, link, onClick, Component, iconUrl, className, isCurrent, isBlank = 'false' }) => {
  const baseClass = 'coop-nav__link'
  const currentClass = isCurrent ? `${baseClass}--is-current ${className}--is-current` : ''
  const allClasses = `${baseClass} ${currentClass} ${className}`

  const content =
    <span>
      <span className={`${className}-name`}>
        {name}
      </span>

      { iconUrl &&
        <img src={getImagePath(iconUrl)} className={`${className}-icon`} />
      }
    </span>

  if (Component) {
    return (
      <div className={allClasses}>
        { Component }
      </div>
    )
  } else if (onClick) {
    return (
      <div onClick={onClick} className={allClasses}>
        { content }
      </div>
    )
  }

  return (
    <a href={link} className={allClasses} target={ isBlank ? '_blank' : '_self' }>
      { content }
    </a>
  )
}

export default NavigationLink
