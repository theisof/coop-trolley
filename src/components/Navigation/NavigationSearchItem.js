import React from 'react'

class NavigationSearch extends React.Component {
  constructor(props) {
    super(props);

    this.state = {}
    this.handleClick = this.handleClick.bind(this)
  }

  handleClick(e) {
    const { onSearchItemClick } = this.props

    onSearchItemClick(this.props)
  }

  render() {
    const { thumbnail, name } = this.props

    return (
      <div className='coop-nav-search__item' onClick={this.handleClick}>
        {thumbnail &&
          <img src={thumbnail} alt={name} className='coop-nav-search__thumbnail' />
        }

        <div  className='coop-nav-search__name'>
          { name }
        </div>
      </div>
    )
  }
}

export default NavigationSearch


/*

   <a href={r.link} key={e} className='coop-nav-search__link'>
        <img src={r.thumbnail} alt={r.name} className='coop-nav-search__thumbnail' />

        <div  className='coop-nav-search__name'>
          { r.name }
        </div>
      </a>
*/