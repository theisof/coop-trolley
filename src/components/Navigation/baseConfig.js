export default {
  siteLogo: {
    imageUrl: null,
    link: '/'
  },
  globalItems: [
    {
      name: 'Kundeservice',
      link: 'https://coop.dk/kundeservice'
    },
    {
      name: 'Bliv medlem',
      link: 'https://tilmeld.coop.dk/standard/forside'
    }
  ],
  siteItems: [],
  contextItems: []
}