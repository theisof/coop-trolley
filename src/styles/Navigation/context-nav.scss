.coop-context-nav {
  background-color: $color-gray-dark;
}

.coop-context-nav__list {
  display: flex;
  justify-content: center;
}

.coop-context-nav__link {
  padding: 15px 20px;
  font-size: 15px;
  letter-spacing: 0.05em;

  &,
  &:visited,
  &:focus {
    color: #fff;
  }

  &:hover,
  &:active {
    color: #fff;
    text-decoration: none;
  }
}

.coop-context-nav__link--is-current {
  &,
  &:visited,
  &:focus,
  &:hover,
  &:active {
    color: $color-primary;
    font-weight: bold;
  }
}

@media (max-width: $breakpoint-sm) {
  .coop-context-nav {
    position: fixed;
    top: $site-nav-height;
    left: 0;
    height: 100%;
    width: $mobile-width;
    overflow-y: scroll;
    -webkit-overflow-scrolling: touch;
    z-index: $z-index-backdrop + 1;
    box-shadow: 3px 0px 10px 0px rgba(0,0,0,0.35);
  }

  .coop-context-nav__list {
    flex-direction: column;
    justify-content: flex-start;
    padding: 0;
  }

  .coop-context-nav__link {
    padding: 20px 25px;
    font-size: 16px;
    border-bottom: 1px solid lighten($color-gray-dark, 5%);
  }
}