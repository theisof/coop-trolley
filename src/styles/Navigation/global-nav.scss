.coop-global-nav {
  background-color: $color-sand;
  justify-content: flex-end;
  display: flex;
  align-items: center;
  height: $global-nav-height;
}

.coop-global-nav__list {
  display: flex;
}

.coop-global-nav__link {
  padding: 5px 0 5px 15px;
  font-size: 12px;
  display: block;
}

.coop-global-nav__close-wrap {
  display: none;
}

@media (max-width: $breakpoint-sm) {
  .coop-global-nav {
    position: fixed;
    top: 0;
    z-index: $z-index-backdrop + 2;
    height: $site-nav-height;
    width: $mobile-width;
    justify-content: flex-start;
    box-shadow: 3px 0px 10px 0px rgba(0,0,0,0.35);
  }

  .coop-global-nav__close-wrap {
    display: flex;
    align-items: center;
    justify-content: center;
    padding: 0 12px;
    height: 100%;
    margin-right: auto;
  }

  .coop-global-nav__close-image {
    height: 20px;
  }

  .coop-global-nav__list {
    padding: 5px 10px;
  }

  .coop-global-nav__item {
    display: none;

    &:nth-last-of-type(-n+2) {
      display: block;
    }
  }

  .coop-global-nav__link {
    white-space: nowrap;
    overflow: hidden;
    text-overflow: ellipsis;
    max-width: 110px;
    padding: 5px;
  }
}
