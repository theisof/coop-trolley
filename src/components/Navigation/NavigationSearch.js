import React from 'react'
import NavigationSearchItem from './NavigationSearchItem'

class NavigationSearch extends React.Component {
  constructor(props) {
    super(props);

    this.state = {}
    this.handleInputChange = this.handleInputChange.bind(this)
  }

  handleInputChange(e) {
    const { onSearchInput } = this.props

    onSearchInput(e.target.value)
  }

  render() {
    const { searchResults, onSearchItemClick } = this.props

    return (
      <div className='coop-nav-search'>
        <input
          className='coop-nav-search__input'
          placeholder="SÃ¸g..."
          onChange={this.handleInputChange}
        />

        <div className='coop-nav-search__results'>
          { searchResults.map((s, i) =>
            <div key={i} className='coop-nav-search__section'>
              <h4 className='coop-nav-search__section-title'>{s.title}</h4>

              <div className='coop-nav-search__section-results'>
                { s.results.map((r, e) =>
                  <NavigationSearchItem key={e} {...r} onSearchItemClick={onSearchItemClick} />
                )}
              </div>
            </div>
          )}
        </div>
      </div>
    )
  }
}

export default NavigationSearch