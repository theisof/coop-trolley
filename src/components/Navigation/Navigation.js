import React from 'react'
import baseConfig from './baseConfig'
import NavigationLink from './NavigationLink'
import Backdrop from '../Backdrop'
import Search from '../Search'
import getImagePath from '../../utils/getImagePath'

class Navigation extends React.Component {
  constructor(props) {
    super(props);

    this.state = {
      mobileNavVisible: false,
      mobileSearchVisible: false,
      navStart: 0,
      isFixed: false
    }

    this.showMobileNav = this.showMobileNav.bind(this)
    this.hideMobileNav = this.hideMobileNav.bind(this)
    this.showMobileSearch = this.showMobileSearch.bind(this)
    this.resetSearch = this.resetSearch.bind(this)
    this.handleKeyPress = this.handleKeyPress.bind(this)
    this.handleScroll = this.handleScroll.bind(this)
    this.siteNav = React.createRef()
  }

  handleKeyPress(e) {
    if(e.keyCode === 27) {
      this.setState({
        mobileNavVisible: false,
        mobileSearchVisible: false,
      })
    }
  }

  componentDidMount(){
    const rect = this.siteNav.current.getBoundingClientRect()
    this.setState({ navStart: rect.top + window.scrollY })

    document.addEventListener("scroll", this.handleScroll, false)
    document.addEventListener("keydown", this.handleKeyPress, false)
  }

  componentWillUnmount(){
    document.removeEventListener("scroll", this.handleScroll, false)
    document.removeEventListener("keydown", this.handleKeyPress, false)
  }

  handleScroll() {
    const { navStart, isFixed } = this.state
    const bodyOffset = parseInt(document.body.style.top, 10) || 0 // top offset will be different from 0 if a modal is open

    if (navStart < window.scrollY && ! isFixed) {
      this.setState({ isFixed: true })
    } else if (navStart > window.scrollY && bodyOffset === 0 && isFixed) {
      this.setState({ isFixed: false })
    }
  }

  showMobileNav () {
    this.setState({ mobileNavVisible: true })
  }

  hideMobileNav () {
    this.setState({ mobileNavVisible: false })
  }

  showMobileSearch () {
    this.setState({ mobileSearchVisible: true })
  }

  resetSearch () {
    this.setState({ mobileSearchVisible: false })
    this.props.onSearchReset()
  }

  render() {
    const { globalItems, siteItems, contextItems, siteLogo } = { ...baseConfig, ...this.props.config }
    const { mobileNavVisible, mobileSearchVisible, isFixed } = this.state
    const mobileNavVisibleClass = mobileNavVisible ? 'coop-nav--mobile-visible' : ''
    const mobileSearchVisibleClass = mobileSearchVisible ? 'coop-site-nav__search--mobile-visible' : ''
    const isFixedClass = isFixed ? 'coop-site-nav__inner--is-fixed' : ''

    return (
      <header className={`coop-nav ${mobileNavVisibleClass}`}>
        <div className='coop-global-nav'>
          <div className='coop-global-nav__close-wrap' onClick={this.hideMobileNav}>
            <img
              src={getImagePath('times-black.svg')}
              className='coop-global-nav__close-image'
              alt="Luk"
            />
          </div>

          <ul className='coop-nav__list coop-global-nav__list'>
            { globalItems.map((item, i) =>
              <li className='coop-global-nav__item' key={i}>
                <NavigationLink {...item} className='coop-global-nav__link' isBlank />
              </li>
            )}
          </ul>
        </div>

        <div className='coop-site-nav' ref={this.siteNav}>
          <div className={`coop-site-nav__inner  ${isFixedClass}`}>
            <div className='coop-site-nav__mobile'>
              <img
                src={getImagePath('burger.svg')}
                alt="menu"
                className='coop-site-nav__mobile-toggle'
                onClick={this.showMobileNav}
              />

              <img
                src={getImagePath('search.svg')}
                alt="sÃ¸g"
                className='coop-site-nav__mobile-search'
                onClick={this.showMobileSearch}
              />
            </div>

            <a href={siteLogo.link} className='coop-site-nav__logo-wrap'>
              <img src={getImagePath(siteLogo.image)} alt="logo" className='coop-site-nav__logo-image' />
            </a>

            <div className={`coop-site-nav__search ${mobileSearchVisibleClass}`}>
              <Search
                {...this.props}
                resetSearch={this.resetSearch}
                mobileSearchVisible={mobileSearchVisible}
              />
            </div>

            <ul className='coop-nav__list coop-site-nav__list'>
              { siteItems.map((item, i) =>
                <li className='coop-site-nav__item' key={i}>
                  <NavigationLink {...item} className='coop-site-nav__link' />
                </li>
              )}
            </ul>
          </div>
        </div>

        <div className='coop-context-nav'>
          <ul className='coop-nav__list coop-context-nav__list'>
            { contextItems.map((item, i) =>
              <li className='coop-context-nav__item' key={i}>
                <NavigationLink {...item} className='coop-context-nav__link' />
              </li>
            )}
          </ul>
        </div>

        <Backdrop visible={mobileNavVisible} onClose={this.hideMobileNav} />
      </header>
    )
  }
}

export default Navigation