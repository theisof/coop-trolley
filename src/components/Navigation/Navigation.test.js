import React from 'react'
import { mount } from 'enzyme'
import toJSON from 'enzyme-to-json';
import Navigation from './Navigation'

it('Navigation renders correctly', () => {
  const tree = mount( <Navigation />)

  expect(toJSON(tree)).toMatchSnapshot()
})
