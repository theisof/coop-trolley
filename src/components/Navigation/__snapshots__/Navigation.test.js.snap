// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Navigation renders correctly 1`] = `
<Navigation>
  <header
    className="coop-nav "
  >
    <div
      className="coop-global-nav"
    >
      <div
        className="coop-global-nav__close-wrap"
        onClick={[Function]}
      >
        <img
          alt="Luk"
          className="coop-global-nav__close-image"
          src="https://cdn.jsdelivr.net/gh/theisof/coop-trolley@2.0.0/images/times-black.svg"
        />
      </div>
      <ul
        className="coop-nav__list coop-global-nav__list"
      >
        <li
          className="coop-global-nav__item"
          key="0"
        >
          <NavigationLink
            className="coop-global-nav__link"
            isBlank={true}
            link="https://coop.dk/kundeservice"
            name="Kundeservice"
          >
            <a
              className="coop-nav__link  coop-global-nav__link"
              href="https://coop.dk/kundeservice"
              target="_blank"
            >
              <span>
                <span
                  className="coop-global-nav__link-name"
                >
                  Kundeservice
                </span>
              </span>
            </a>
          </NavigationLink>
        </li>
        <li
          className="coop-global-nav__item"
          key="1"
        >
          <NavigationLink
            className="coop-global-nav__link"
            isBlank={true}
            link="https://tilmeld.coop.dk/standard/forside"
            name="Bliv medlem"
          >
            <a
              className="coop-nav__link  coop-global-nav__link"
              href="https://tilmeld.coop.dk/standard/forside"
              target="_blank"
            >
              <span>
                <span
                  className="coop-global-nav__link-name"
                >
                  Bliv medlem
                </span>
              </span>
            </a>
          </NavigationLink>
        </li>
      </ul>
    </div>
    <div
      className="coop-site-nav"
    >
      <div
        className="coop-site-nav__inner  "
      >
        <div
          className="coop-site-nav__mobile"
        >
          <img
            alt="menu"
            className="coop-site-nav__mobile-toggle"
            onClick={[Function]}
            src="https://cdn.jsdelivr.net/gh/theisof/coop-trolley@2.0.0/images/burger.svg"
          />
          <img
            alt="søg"
            className="coop-site-nav__mobile-search"
            onClick={[Function]}
            src="https://cdn.jsdelivr.net/gh/theisof/coop-trolley@2.0.0/images/search.svg"
          />
        </div>
        <a
          className="coop-site-nav__logo-wrap"
          href="/"
        >
          <img
            alt="logo"
            className="coop-site-nav__logo-image"
            src="https://cdn.jsdelivr.net/gh/theisof/coop-trolley@2.0.0/images/undefined"
          />
        </a>
        <div
          className="coop-site-nav__search "
        >
          <Search
            hideSearch={[Function]}
            mobileSearchVisible={false}
          >
            <div
              className="coop-search"
            >
              <img
                alt="søg"
                className="coop-search__icon"
                src="https://cdn.jsdelivr.net/gh/theisof/coop-trolley@2.0.0/images/search.svg"
              />
              <input
                className="coop-search__input"
                id="coop-search__input"
                onChange={[Function]}
                placeholder="Søg..."
                value=""
              />
              <div
                className="coop-search__results"
                style={Object {}}
              />
              <div
                className="coop-search__clear-wrap "
                onClick={[Function]}
              >
                <img
                  alt="Luk"
                  className="coop-search__clear-image"
                  src="https://cdn.jsdelivr.net/gh/theisof/coop-trolley@2.0.0/images/times-black.svg"
                />
              </div>
            </div>
            <Backdrop
              onClose={[Function]}
              visible={false}
            >
              <div
                className="coop-backdrop "
                onClick={[Function]}
                style={
                  Object {
                    "zIndex": 100000,
                  }
                }
              />
            </Backdrop>
          </Search>
        </div>
        <ul
          className="coop-nav__list coop-site-nav__list"
        />
      </div>
    </div>
    <div
      className="coop-context-nav"
    >
      <ul
        className="coop-nav__list coop-context-nav__list"
      />
    </div>
    <Backdrop
      onClose={[Function]}
      visible={false}
    >
      <div
        className="coop-backdrop "
        onClick={[Function]}
        style={
          Object {
            "zIndex": 100000,
          }
        }
      />
    </Backdrop>
  </header>
</Navigation>
`;
